edweiss-app/components/lectures/slides/StudentQuestion.tsx

```diff
@@ -1,55 +1,146 @@
+                });
+            }
+        } catch (error) {
+            // Display feedback to the user when error adding question
+            Toast.show({
+                type: 'error',
+                text1: 'Your message submition encountered an error: ',
+                text2: error instanceof Error ? error.message : JSON.stringify(error), // Include the error details
+            });
+        } finally {
+            setIsLoading(false);
+        }
+    }
+
+    // Function for updating a question 
+    async function updateQuestion(id: string, likes: number) {
+        try {
+            const res = await callFunction(LectureDisplay.Functions.updateQuestion, {
                 courseId: courseName,
                 lectureId: lectureId,
-                question: question
+                id: id,
+                likes: likes,
             });
+            if (!res.status) {
+                console.log(res.error)
+                // Display feedback to the user when failure (empty question)
+                Toast.show({
+                    type: 'error',
+                    text1: 'You were unable to like/unlike this message',
+                });
+            }
         } catch (error) {
-            console.error("Error creating question:", error);
+            // Display feedback to the user when failing to like question
+            Toast.show({
+                type: 'error',
+                text1: 'Your like/unlike attempt encountered an error: ',
+                text2: error instanceof Error ? error.message : JSON.stringify(error), // Include the error details
+            });
         }
-        setQuestion('');
-        setIsLoading(false);
+    }
+
+    const QuestionItem: React.FC<{
+        question: Document<LectureDisplay.Question>;
+        index: number;
+    }> = ({ question, index }) => {
+        const { text, anonym, userID, likes, username } = question.data;
+        const isUser = uid == userID;
+        const id = question.id;
+
+        return (
+            <TView key={index} mb={'sm'} backgroundColor={isUser ? 'sapphire' : 'crust'} borderColor='surface0' radius={'lg'} flex={1} flexDirection='column' ml='sm' style={{ right: isUser ? 0 : 10 }}>
+                <TText ml={16} mb={4} size={'sm'} pl={2} pt={'sm'} color='overlay2'>{anonym ? "Anonym" : username}</TText>
+
+                <TView pr={'sm'} pl={'md'} pb={'sm'} flexDirection='row' justifyContent='space-between' alignItems='flex-start'>
+                    <TText ml={10} color='overlay0'>{text}</TText>
+                    <TView pr={'sm'} pl={'md'} pb={'sm'} flexDirection='row' alignItems='flex-end'>
+                        <TText color='text'>{likes}</TText>
+                        {!isUser && <TTouchableOpacity testID={`like-button-${index}`} backgroundColor='transparent' onPress={() => {
+                            if (SyncStorage.get(id) !== undefined) {
+                                SyncStorage.set(id, !SyncStorage.get(id));
```

Comment:
Try to avoid using unnamed `id` directly in sync storage, as it might clash with other parts. Try to use maybe `stquestion-${id}` if you really need to use `SyncStorage`.

Reply:
I am not interely sure I understood this, you mean that the name "id" that I just used could cause bugs due to it being a name that should be avoided since it could refer to other values? because I initialized this constant in the beginning of that function with 
 `const id = question.id;`
so you are just saying for me to use question.id directly or rename the constant to another name?

Reply:
Hello,
Instead of
```typescript
SyncStorage.set(id, ...);
```
Do
```typescript
SyncStorage.set(`stquestion-${id}`, ...);
```
It's to not have any clash with any other module, with `SyncStorage`, we try to use human friendly explicit names like that.

Reply:
Ok I understand it now thank you so much